dashboard/app/models/stage.rb

```diff
@@ -37,8 +37,16 @@ def unplugged?
   end
 
   def localized_title
+    # The standard case for localized_title is something like "Stage 1: Maze".
+    # In the case of lockable stages, we don't want to include the Stage 1
+    return I18n.t("data.script.name.#{script.name}.#{name}") if lockable
+
     if script.stages.to_a.many?
-      I18n.t('stage_number', number: position) + ': ' + I18n.t("data.script.name.#{script.name}.#{name}")
+      # Because lockable stages aren't numbered, our stage number is actually our
+      # position, minus the number of lockable stages preceeding us
+      stage_number = position - script.stages.to_a[0, position].count(&:lockable)
```

Comment:
Could cache `total_lockable`, otherwise we'll be counting lockable `n` times where `n` is the number of stages in the script.  But it's on the cached array, so not a big issue.